lib/datadog/tracing/span_operation.rb

```diff
@@ -352,17 +352,23 @@ class Events
         attr_reader \
           :after_finish,
           :after_stop,
-          :before_start,
-          :on_error
+          :before_start
 
         def initialize(on_error: nil)
           @after_finish = AfterFinish.new
           @after_stop = AfterStop.new
           @before_start = BeforeStart.new
-          @on_error = OnError.new
+        end
 
-          # Set default error behavior
-          @on_error.subscribe(:default, &DEFAULT_ON_ERROR)
+        # This event is lazily initialized as error paths
+        # are normally less common that non-error paths.
+        def on_error
+          @on_error ||= begin
+            event = OnError.new
+            # Set default error behavior
+            event.subscribe(:default, &DEFAULT_ON_ERROR)
+            event
+          end
```

Comment:
This is not thread-safe; is that a problem? In particular, the `Event` itself seems to be built to be thread-safe, so I just wanted to doublecheck what our expectations are around this `Events` class.

Reply:
Ah, I guess https://github.com/DataDog/dd-trace-rb/pull/1890 is partially my answer?

Reply:
> Ah, I guess #1890 is partially my answer?

Yes :)